File: modules/react-native-mlkit-core/ios/RNMLKitRect.swift
import ExpoModulesCore

public struct RNMLKitRect: Record {
    public init(){}
    
    @Field
    var origin: RNMLKitPoint = RNMLKitPoint()
    @Field
    var size: RNMLKitPoint = RNMLKitPoint()

    public static func fromCGRect(rect: CGRect) -> RNMLKitRect {
        let expoRect = RNMLKitRect()

        let origin = RNMLKitPoint()
        origin.x = rect.origin.x
        origin.y = rect.origin.y
        
        let size = RNMLKitPoint()
        size.x = rect.width
        size.y = rect.height
        
        expoRect.origin = origin;
        expoRect.size = size
        
        return expoRect
    }
}

File: modules/react-native-mlkit-core/android/src/main/java/red/infinite/reactnativemlkit/core/RNMLKitRect.kt
package red.infinite.reactnativemlkit.core

import android.graphics.Rect
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

data class RNMLKitRect(
    @Field var origin: RNMLKitPoint,
    @Field var size: RNMLKitPoint
) : Record {
    companion object {
        fun fromRect(rect: Rect): RNMLKitRect {
            return RNMLKitRect(
                origin = RNMLKitPoint(rect.left.toFloat(), rect.top.toFloat()),
                size = RNMLKitPoint(rect.width().toFloat(), rect.height().toFloat())
            )
        }

        fun zero(): RNMLKitRect {
            return RNMLKitRect(
                origin = RNMLKitPoint.zero(),
                size = RNMLKitPoint.zero()
            )
        }
    }


}
